### PR TITLE
setuptools/dist: Fix reproducibility issue by sorting globbing

### PR DESCRIPTION
**Pull request by rpurdie**
globbing uses os.listdir() which returns order on disk which is unsorted.
Ultimately this leads to random ordering of License-File entries in PKG-INFO
and means the resulting output isn't deterministic.

Adding a sort fixes that. Switch to glob instead ot iglob since sorted()
removes the benefit of an iterator.

https://github.com/pypa/setuptools/issues/2691

Signed-off-by: Richard Purdie <richard.purdie@linuxfoundation.org>

## Summary of changes

Closes 

### Pull Request Checklist
- [x] Changes have tests
- [x] News fragment added in [`changelog.d/`].
  _(See [documentation][PR docs] for details)_

[`changelog.d/`]: https://github.com/pypa/setuptools/tree/master/changelog.d
[PR docs]:
https://setuptools.readthedocs.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

**Comment by jaraco** [comment]
> Switch to glob instead ot iglob since sorted()
> removes the benefit of an iterator.

I think I'd still prefer an iterator.

Can you create a test to capture this expectation?

**Review by webknjaz — COMMENTED** [review]
Could you add a change not file following this instruction? https://setuptools.readthedocs.io/en/latest/development/developer-guide.html#adding-change-notes-with-your-prs

File: setuptools/dist.py [hunk 1]
```diff
@@ -603,7 +603,7 @@ def _expand_patterns(patterns):
         return (
```
Comment (webknjaz) [inline setuptools/dist.py:603]: If you want reproducibility, why not sort the filtered result? Of course, it'll turn a generator expression into a list or something...

Comment (jaraco) [reply setuptools/dist.py:603]: I prefer this minimal approach that retains the order of each pattern. If you were to sort the entire result, you would lose fidelity about the order in which the patterns were supplied. Thus if the user specifies LICENSE_B, LICENSE_A as separate globs, they'll get them in that order.

**Pull request merged**